nilearn/surface/surface.py

```diff
@@ -0,0 +1,644 @@
+        How many samples are drawn around each vertex and averaged. If
+        ``None``, use a reasonable default for the chosen sampling strategy
+        (20 for 'ball' or 10 for 'line').
+        For performance reasons, if using `kind` ="ball", choose `n_samples` in
+        [10, 20, 40, 80, 160] (default is 20), because cached positions are
+        available.
+
+    mask_img : Niimg-like object or None, optional (default=None)
+        Samples falling out of this mask or out of the image are ignored.
+        If ``None``, don't apply any mask.
+
+    Returns
+    -------
+    texture : numpy.ndarray, 1d or 2d.
+        If 3D image is provided, a 1d vector is returned, containing one value
+        for each mesh node.
+        If 4D image is provided, a 2d array is returned, where each row
+        corresponds to a mesh node.
+
+    Notes
+    -----
+    This function computes a value for each vertex of the mesh. In order to do
+    so, it selects a few points in the volume surrounding that vertex,
+    interpolates the image intensities at these sampling positions, and
+    averages the results.
+
+    Two strategies are available to select these positions.
+        - 'ball' uses points regularly spaced in a ball centered at the mesh
+            vertex. The radius of the ball is controlled by the parameter
+            `radius`.
+        - 'line' starts by drawing the normal to the mesh passing through this
+            vertex. It then selects a segment of this normal, centered at the
+            vertex, of length 2 * `radius`. Image intensities are measured at
+            points regularly spaced on this normal segment.
+
+    You can control how many samples are drawn by setting `n_samples`.
+
+    Once the sampling positions are chosen, those that fall outside of the 3d
+    image (or ouside of the mask if you provided one) are discarded. If all
+    sample positions are discarded (which can happen, for example, if the
+    vertex itself is outside of the support of the image), the projection at
+    this vertex will be ``numpy.nan``.
+    The 3d image then needs to be interpolated at each of the remaining points.
+    Two options are available: 'nearest' selects the value of the nearest
+    voxel, and 'linear' performs trilinear interpolation of neighbouring
+    voxels. 'linear' may give better results - for example, the projected
+    values are more stable when resampling the 3d image or applying affine
+    transformations to it. 'nearest' is slightly faster for one image, and much
+    faster for many images (can be an order of magnitude faster for 100 images,
+    i.e. a 4d niimg with last dimension=100). Note that linear interpolation is
+    only available if your scipy version is 0.14 or more recent.
+
+    Once the 3d image has been interpolated at each sample point, the
+    interpolated values are averaged to produce the value associated to this
+    particular mesh vertex.
+
+    """
```

Comment:
I would like here a warning that says that the function is experimental and that minor details such in the interpolation could change.

The reason that I would like this, is that we are considering merging this just before a release, and hence we have little insight on the best choices for hyperparameters. I want to be able to change those hyperparameters.

We'll remove the warning in the near future.